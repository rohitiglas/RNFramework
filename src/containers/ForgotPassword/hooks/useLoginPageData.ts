import {useCallback} from 'react';
import {useFormik} from 'formik';
import {LoginFormModel} from '../types';
import {validationSchema} from '../validation';

export const useLoginPageData = () => {
  const onSubmit = useCallback(() => {}, []);

  const formik = useFormik<LoginFormModel>({
    initialValues: {
      email: '',
      password: '',
    },
    validationSchema,
    onSubmit,
    enableReinitialize: false,
    validateOnChange: true,
    validateOnMount: true,
  });

  return {
    formik,
  };
};
