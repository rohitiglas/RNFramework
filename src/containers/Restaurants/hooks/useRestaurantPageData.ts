import {useCallback, useEffect} from 'react';
import {useFormik} from 'formik';
import {useDispatch, useSelector} from 'react-redux';
import {LoginFormModel} from '../types';
import {validationSchema} from '../validation';
import {fetchData} from '../../../store/actions/swiggyAction';

export const useRestaurantPageData = () => {
  const dispatch = useDispatch();
  const {data} = useSelector(state => state.data);
  useEffect(() => {
    dispatch(fetchData());
  }, [dispatch]);
  const onSubmit = useCallback(() => {}, []);

  const formik = useFormik<LoginFormModel>({
    initialValues: {
      email: '',
      password: '',
    },
    validationSchema,
    onSubmit,
    enableReinitialize: false,
    validateOnChange: true,
    validateOnMount: true,
  });

  return {
    formik,
    data,
  };
};
