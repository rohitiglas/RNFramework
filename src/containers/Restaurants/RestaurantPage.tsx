import React, {memo} from 'react';
import {FormikProvider} from 'formik';

import {useRestaurantPageData} from './hooks/useRestaurantPageData';
import RestaurantTemplate from '../../presentational/components/templates/RestaurantTemplate/RestaurantTemplate';

export const RestaurantPage = memo(() => {
  const {formik, data} = useRestaurantPageData();

  return (
    <FormikProvider value={formik}>
      <RestaurantTemplate restaurantData={data} />
    </FormikProvider>
  );
});
