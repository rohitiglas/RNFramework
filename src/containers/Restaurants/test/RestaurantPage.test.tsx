import React from 'react';
import {render} from '@testing-library/react-native';
import {FormikProvider} from 'formik'; // Import FormikProvider
import {useRestaurantPageData} from '../hooks/useRestaurantPageData'; // Adjust the import path as needed
import {RestaurantPage} from '../RestaurantPage'; // Import your component

// Mock the useRestaurantPageData hook to provide test data
jest.mock('../hooks/useRestaurantPageData', () => ({
  useRestaurantPageData: jest.fn(),
}));

describe('RestaurantPage', () => {
  it('renders RestaurantTemplate with provided data', () => {
    // Define mock data for your test
    const mockData = [
      {
        info: {
          id: '565266',
          name: 'Om Bikaner Sweets & Snacks',
          cloudinaryImageId: 'eu55r6igxo77ahgumivy',
          locality: 'Sinda Chowk',
          areaName: 'Sector 12 A',
          costForTwo: '₹200 for two',
          cuisines: ['Snacks', 'Chinese', 'South Indian'],
          avgRating: 3.4,
          veg: true,
          feeDetails: {
            restaurantId: '565266',
            fees: [
              {
                name: 'BASE_DISTANCE',
                fee: 3800,
              },
              {
                name: 'BASE_TIME',
              },
              {
                name: 'ANCILLARY_SURGE_FEE',
              },
            ],
            totalFee: 3800,
          },
          parentId: '151508',
          avgRatingString: '3.4',
          totalRatingsString: '100+',
          sla: {
            deliveryTime: 20,
            lastMileTravel: 2.3,
            serviceability: 'SERVICEABLE',
            slaString: '20 mins',
            lastMileTravelString: '2.3 km',
            iconType: 'ICON_TYPE_EMPTY',
          },
          availability: {
            nextCloseTime: '2023-09-27 21:16:00',
            opened: true,
          },
          badges: {},
          isOpen: true,
          type: 'F',
          badgesV2: {
            entityBadges: {
              imageBased: {},
              textBased: {},
              textExtendedBadges: {},
            },
          },
          aggregatedDiscountInfoV3: {
            header: '10% OFF',
            subHeader: 'UPTO ₹40',
          },
          orderabilityCommunication: {
            title: {},
            subTitle: {},
            message: {},
            customIcon: {},
          },
          differentiatedUi: {
            displayType: 'ADS_UI_DISPLAY_TYPE_ENUM_DEFAULT',
            differentiatedUiMediaDetails: {
              mediaType: 'ADS_MEDIA_ENUM_IMAGE',
              lottie: {},
              video: {},
            },
          },
          reviewsSummary: {},
          displayType: 'RESTAURANT_DISPLAY_TYPE_DEFAULT',
          restaurantOfferPresentationInfo: {},
        },
        analytics: {
          context: 'seo-data-a03d3313-a353-4ff2-9809-711f4721d02d',
        },
        cta: {
          link: 'https://www.swiggy.com/restaurants/om-bikaner-sweets-and-snacks-sinda-chowk-sector-12-a-gurgaon-565266',
          text: 'RESTAURANT_MENU',
          type: 'WEBLINK',
        },
        widgetId: 'collectionV5RestaurantListWidget_SimRestoRelevance_food_seo',
      },
    ];

    // Mock the useRestaurantPageData hook to return your mock data
    useRestaurantPageData.mockReturnValue({formik: {}, data: mockData});

    // Render the RestaurantPage component within the FormikProvider
    const page = render(
      <FormikProvider>
        <RestaurantPage />
      </FormikProvider>,
    );

    expect(page).toMatchSnapshot();
  });

  // Add more test cases as needed
});
