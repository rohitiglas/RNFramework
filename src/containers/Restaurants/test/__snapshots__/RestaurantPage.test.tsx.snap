// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantPage renders RestaurantTemplate with provided data 1`] = `
<View
  style={
    {
      "flex": 1,
      "marginTop": 0,
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "analytics": {
            "context": "seo-data-a03d3313-a353-4ff2-9809-711f4721d02d",
          },
          "cta": {
            "link": "https://www.swiggy.com/restaurants/om-bikaner-sweets-and-snacks-sinda-chowk-sector-12-a-gurgaon-565266",
            "text": "RESTAURANT_MENU",
            "type": "WEBLINK",
          },
          "info": {
            "aggregatedDiscountInfoV3": {
              "header": "10% OFF",
              "subHeader": "UPTO ₹40",
            },
            "areaName": "Sector 12 A",
            "availability": {
              "nextCloseTime": "2023-09-27 21:16:00",
              "opened": true,
            },
            "avgRating": 3.4,
            "avgRatingString": "3.4",
            "badges": {},
            "badgesV2": {
              "entityBadges": {
                "imageBased": {},
                "textBased": {},
                "textExtendedBadges": {},
              },
            },
            "cloudinaryImageId": "eu55r6igxo77ahgumivy",
            "costForTwo": "₹200 for two",
            "cuisines": [
              "Snacks",
              "Chinese",
              "South Indian",
            ],
            "differentiatedUi": {
              "differentiatedUiMediaDetails": {
                "lottie": {},
                "mediaType": "ADS_MEDIA_ENUM_IMAGE",
                "video": {},
              },
              "displayType": "ADS_UI_DISPLAY_TYPE_ENUM_DEFAULT",
            },
            "displayType": "RESTAURANT_DISPLAY_TYPE_DEFAULT",
            "feeDetails": {
              "fees": [
                {
                  "fee": 3800,
                  "name": "BASE_DISTANCE",
                },
                {
                  "name": "BASE_TIME",
                },
                {
                  "name": "ANCILLARY_SURGE_FEE",
                },
              ],
              "restaurantId": "565266",
              "totalFee": 3800,
            },
            "id": "565266",
            "isOpen": true,
            "locality": "Sinda Chowk",
            "name": "Om Bikaner Sweets & Snacks",
            "orderabilityCommunication": {
              "customIcon": {},
              "message": {},
              "subTitle": {},
              "title": {},
            },
            "parentId": "151508",
            "restaurantOfferPresentationInfo": {},
            "reviewsSummary": {},
            "sla": {
              "deliveryTime": 20,
              "iconType": "ICON_TYPE_EMPTY",
              "lastMileTravel": 2.3,
              "lastMileTravelString": "2.3 km",
              "serviceability": "SERVICEABLE",
              "slaString": "20 mins",
            },
            "totalRatingsString": "100+",
            "type": "F",
            "veg": true,
          },
          "widgetId": "collectionV5RestaurantListWidget_SimRestoRelevance_food_seo",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "margin": 5,
                "width": 220,
              }
            }
            testID="restaurant-item"
          >
            <Image
              source={
                {
                  "uri": "https://media-assets.swiggy.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_660/eu55r6igxo77ahgumivy",
                }
              }
              style={
                {
                  "borderRadius": 20,
                  "height": 200,
                  "width": 200,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                {
                  "color": "rgba(2, 6, 12, 0.75)",
                  "display": "flex",
                  "flexDirection": "column",
                  "flexWrap": "wrap",
                  "fontFamily": "Basis Grotesque Pro",
                  "fontSize": 18,
                  "fontWeight": "700",
                  "letterSpacing": -0.3,
                  "lineHeight": 24,
                  "margin": 0,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              Om Bikaner Sweets & Snacks
            </Text>
            <Text>
              * 
              3.4
            </Text>
            <Text
              style={
                {
                  "color": "rgba(2, 6, 12, 0.6)",
                  "display": "flex",
                  "flexDirection": "column",
                  "flexWrap": "wrap",
                  "fontFamily": "Basis Grotesque Pro",
                  "fontSize": 16,
                  "fontWeight": "200",
                  "letterSpacing": -0.3,
                  "lineHeight": 19,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              Snacks , Chinese , South Indian
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
